pkg/agent/attestor/workload/workload.go

```diff
@@ -0,0 +1,91 @@
+package attestor
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spiffe/spire/pkg/agent/catalog"
+	"github.com/spiffe/spire/pkg/common/telemetry"
+	"github.com/spiffe/spire/proto/agent/workloadattestor"
+	"github.com/spiffe/spire/proto/common"
+)
+
+type attestor struct {
+	c *Config
+}
+
+type Attestor interface {
+	Attest(pid int32) []*common.Selector
+}
+
+func New(config *Config) Attestor {
+	return &attestor{c: config}
+}
+
+type Config struct {
+	Catalog catalog.Catalog
+	L       logrus.FieldLogger
+	T       telemetry.Sink
+}
+
+// Attest invokes all workload attestor plugins against the provided PID. If an error
+// is encountered, it is logged and selectors from the failing plugin are discarded.
+func (wla *attestor) Attest(pid int32) []*common.Selector {
+	tLabels := []telemetry.Label{{"workload_pid", string(pid)}}
```

Comment:
all PIDs are allow? any kind of validation to add here?
how about move "workload_pid", "workload_api" and "workload_attestation_duration" to constants?

Reply:
This function is exposed to retrieve all the selectors for the PID provided. Is responsibility of the caller to get and provide a valid PID. An invalid PID will just return no selectors.